components/Ink/src/private/MDCLegacyInkLayer.m

```diff
@@ -240,6 +245,12 @@ - (void)enterWithCompletion:(void (^)(void))completionBlock {
     [self addAnimation:_foregroundPositionAnim forKey:kInkLayerForegroundPositionAnim];
   }
 
+  if (_foregroundOpacityAnim.duration < _foregroundScaleAnim.duration) {
+    _foregroundScaleAnim.delegate = self;
```

Comment:
We have to add some more delegates for all the animations otherwise we would not get any animation callbacks for them. Also relates all other animation delegate additions in this PR